From 528e5697cb9fada883f55414362adff5f4979e3f Mon Sep 17 00:00:00 2001
From: Li Jun <jun.li@freescale.com>
Date: Fri, 23 Jan 2015 18:35:35 +0800
Subject: [PATCH] MLK-10132-3 usb: chipidea: udc: do not enter low power mode
 if vbus on

This patch is to prevent usb entering low power mode if vbus is on even gadget
driver is not binded, by holding the PM count of ci->dev.
So, there are 3 pm usage_count status:
- ci->dev: 1 ci->gadget.dev: 1
  Device mode with gadget driver binded and vbus on.
- ci->dev: 1 ci->gadget.dev: 0
  USB vbus on but gadget driver not binded.
- ci->dev: 0 ci->gadget.dev: 1
  USB OTG FSM is in a_peripheral mode.
Above 2 device's pm usage_count hold by ci otg(ci->dev) and usb gadget
(ci->gadget.dev).

Signed-off-by: Li Jun <jun.li@freescale.com>
---
 drivers/usb/chipidea/udc.c | 9 +++++++--
 1 file changed, 7 insertions(+), 2 deletions(-)

diff --git a/drivers/usb/chipidea/udc.c b/drivers/usb/chipidea/udc.c
index 4743c6e..8a0724d 100644
--- a/drivers/usb/chipidea/udc.c
+++ b/drivers/usb/chipidea/udc.c
@@ -1865,8 +1865,10 @@ int ci_usb_charger_connect(struct ci_hdrc *ci, int is_active)
 {
 	int ret = 0;
 
+	if (is_active)
+		pm_runtime_get_sync(ci->dev);
+
 	if (ci->platdata->notify_event) {
-		pm_runtime_get_sync(&ci->gadget.dev);
 		if (is_active)
 			hw_write(ci, OP_USBCMD, USBCMD_RS, 0);
 
@@ -1881,8 +1883,11 @@ int ci_usb_charger_connect(struct ci_hdrc *ci, int is_active)
 			/* Pull down dp */
 			hw_write(ci, OP_USBCMD, USBCMD_RS, 0);
 		}
-		pm_runtime_put_sync(&ci->gadget.dev);
 	}
+
+	if (!is_active)
+		pm_runtime_put_sync(ci->dev);
+
 	return ret;
 }
 
-- 
1.8.0

